### PR TITLE
Allow selecting forums in addition to Index and Everywhere

### PR DESCRIPTION
**Pull request by iMattPro**

### CODE REVIEW

File: adm/style/settings_list.html [hunk 1]
```diff
@@ -41,7 +41,21 @@ <h1>{{ lang('BOARD_ANNOUNCEMENTS_SETTINGS') }}</h1>
 	{% for ba in announcements %}
 		<tr>
 			<td>{{ ba.DESCRIPTION }}</td>
-			<td>{{ ba.INDEX_ONLY ? lang('INDEX') : lang('BOARD_ANNOUNCEMENTS_EVERYWHERE') }}</td>
+			<td>
+				{% if ba.LOCATIONS is empty %}
+					{{ lang('BOARD_ANNOUNCEMENTS_EVERYWHERE') }}
+				{% else %}
+					{% set has_index = constant('\\phpbb\\boardannouncements\\ext::INDEX_ONLY') in ba.LOCATIONS %}
```
Comment (rxu) [inline adm/style/settings_list.html:48]: How does `{% set ... in ... %}` construction work? Couldn't find it in docs.

Comment (iMattPro) [reply adm/style/settings_list.html:48]: `set` defines a new variable. https://twig.symfony.com/doc/2.x/tags/set.html
`in` is like in array https://twig.symfony.com/doc/2.x/templates.html#containment-operator

Comment (rxu) [reply adm/style/settings_list.html:48]: Yeah, I see what do they mean individually but just curious how would it work together. So it adds `has_index` element into `ba.LOCATIONS` array?

Comment (iMattPro) [reply adm/style/settings_list.html:48]: No it creates a new twig template variable called `has_index` that is true when `ba.LOCATIONS` contains `-1` which is then used in the next few lines

Comment (iMattPro) [reply adm/style/settings_list.html:48]: it's same as doing:
```php
$has_index = in_array(ext::INDEX_ONLY, $locations);
```

**Comment by rxu** [comment]
May be add user_id to cookie to distinct users dismission?

**Comment by iMattPro** [comment]
The cookie is to handle ANONYMOUS users

**Comment by rxu** [comment]
Then it shouldn't affect registered users, should it? But it does. Or does registered users dismission uses both cookie and DB part? Or am I missing the whole point?

**Comment by iMattPro** [comment]
if a user closes an announcement, it never even gets queried again for them. so the cookie only affects them after they log out.  if they switch to another account the cookie will keep it closed, but in what use case do different people use the same forum on the same device?

**Review by rxu — APPROVED** [review]

**Pull request merged**